File: C++/Labs/Lab5/UpperRegister.cpp
#include "UpperRegister.h"
UpperRegister::UpperRegister(string str)
{
    if (str != "")
    {
        for (size_t i = 0; i < str.length(); i++)
        {
            Str += toupper(str[i]);
        }
    }
}
void UpperRegister::PasteSymbol(char symb) { 
    Str += "/";
    Str += toupper(symb);
}
int UpperRegister::GetLength()
{
    int counter = 0;
    if (Str != "")
    {
        for (int i = 0; i < Str.length(); i++)
        {
            if (Str[i] != '/') counter++;
        }
    }

    return counter;
}

File: CSharp/CSharp/Lab5/UpperRegister.cs
using System;

namespace Lab5
{
    public class UpperRegister : String
    {
        public UpperRegister(string str):base(str)
        {
            Str = Str?.ToUpper();
        }
        public override void PasteSymbol()
        {
            if (Str != null)
            {
                for (int i = 1; i < Str.Length; i += 2)
                {
                    Str = Str.Insert(i, "/");
                }
            }
            else
            {
                Str += "/";
            }
        }
        public override int GetLength()
        {
            int counter = 0;
            if (Str != null)
            {
                foreach (var symb in Str)
                {
                    if (symb != '/') counter++;
                }
            }

            return counter;
        }
    }
}